src/controllers/resource-store/index.tsx

```diff
@@ -70,9 +82,19 @@ export const actions: Actions = {
         ...slice,
         data: getNewSliceData(slice.data),
         expiresAt: getExpiresAt(maxAge),
+        accessedAt: getAccessedAt(),
       })
     );
   },
+  updateRemoteResourceState: (type, key, state) => ({ dispatch, getState }) => {
```

Comment:
This was an edge scenario where, `maxCache` is 1 and we are moving from `page-1` to `page-2` and before the resource request gets complete we switch to `page-2` to `page-3`. In this case `page-2` key would be removed and we don't have to store this key after the resource request is complete. Hence this new function is required which checks if the resource `resourceDataForType[key]` exists only then store the data into the store.

Reply:
What blocks us from adding the check in `setResourceState` directly? Just wondering about the need of a dedicated function "wrapper"

Reply:
It cannot be added to `setResourceState` because it is used for added a new resource as well which doesn't exist yet it the `type`. But in case of `updateRemoteResourceState`, we would be adding a `resource` only when it exists in `type` and not otherwise.